patch_rebaser/patch_rebaser.py

```diff
@@ -228,10 +228,18 @@ def perform_rebase(self):
         try:
             LOGGER.info("Rebasing %s to %s", self.branch, self.commit)
             self.repo.branch.rebase_to_hash(self.branch, self.commit)
-        except git_exceptions.RebaseException:
-            LOGGER.info("Could not rebase. Cleaning up.")
-            self.repo.branch.abort_rebase()
-            raise
+        except git_exceptions.RebaseException as e:
+            if '.gitreview' in str(e):
+                LOGGER.warning("A patch including the .gitreview file failed "
+                               "to rebase, skipping it.")
+                self.repo.git.rebase('--skip')
+                self.perform_rebase()   # Just in case, rebase again
+                LOGGER.info("Re-creating .gitreview change")
+                self._rebuild_gitreview()
```

Comment:
Actually, just had another thought that would make this chunk feel (to me) a bit less intertwined.  What if we move out the git.rebase and perform_rebase calls out of this exception, and into _rebuild_gitreview?  then it feels less like we are doing extra work within the exception itself, even though perhaps technically it still is.  Another option would be to extend RebaseException to handle this code (which could be done a couple ways if we went that path).  However, if everyone else is ok with this, I am willing to let it be as is for now and consider other implementation later.